### PR TITLE
Add js string codebytes

### PR DESCRIPTION
**Pull request by mehboobali98**
### Description

This closes (https://github.com/Codecademy/docs/issues/2513)
Added codebyte examples to string .charAt(), .concat() and .indexOf() methods.

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/javascript/concepts/strings/terms/charAt/charAt.md [hunk 1]
```diff
@@ -45,3 +45,17 @@ const lastLetter = greeting.charAt(greeting.length - 1);
...
+The following is runnable, and demonstrates the use of the `.charAt()` method:
+
+```codebyte/javascript
+const myString = 'I love JavaScript! ';
+
+// Using integer value
+console.log(myString.charAt(2));
+
+// Using decimal value that gets rounded down from 3.5 to 3
+console.log(myString.charAt(2));
```
Comment (SSwiniarski) [inline content/javascript/concepts/strings/terms/charAt/charAt.md:60]:
```suggestion
console.log(myString.charAt(3.5));
```

File: content/javascript/concepts/strings/terms/indexOf/indexOf.md [hunk 2]
```diff
@@ -70,3 +70,27 @@ const didYouFindWaldo = people.indexOf('Waldo');
...
+console.log(myString.indexOf('hate'));
+
+// Using an array
+const myArray = ['Javascript, Python, Ruby, GO'];
+
+// Case 1: Searchable value exists
+console.log(myArray.indexOf('GO'));
+
+// Case 2: Searchable value does not exist
+console.log(myArray.indexOf('C++'));
```
Comment (SSwiniarski) [inline content/javascript/concepts/strings/terms/indexOf/indexOf.md:95]:
```suggestion
```
The `array.indexOf()` method is different from the `string.indexOf()` method and has its own entry here: https://www.codecademy.com/resources/docs/javascript/arrays/indexOf

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@mehboobali98 I did a first review and left a couple of comments.

File: content/javascript/concepts/strings/terms/indexOf/indexOf.md (hunk 2 above)
Comment (mehboobali98) [reply content/javascript/concepts/strings/terms/indexOf/indexOf.md:95]: Removed.

**Comment by mehboobali98** [comment]
@SSwiniarski Resolved the comments.

**Review by SSwiniarski — APPROVED** [review]
@mehboobali98 looks good for a second review.

**Review by caupolicandiaz — APPROVED** [review]
@mehboobali98 @SSwiniarski I've added a few additional edits to these entries. Looks good to go.

**Comment by SSwiniarski** [comment]
@mehboobali98 congrats on the new edits!

Live links:
- https://www.codecademy.com/resources/docs/javascript/strings/charAt
- https://www.codecademy.com/resources/docs/javascript/strings/concat
- https://www.codecademy.com/resources/docs/javascript/strings/indexOf

**Pull request merged**